instance/models/utils.py

```diff
@@ -423,9 +432,48 @@ class ConsulAgent(object):
     managing prefixes, and reduces the call size to the main needed things with a
     possibility to expand it for more advanced queries.
     """
-    def __init__(self, prefix=''):
+
+    # We use the `consul lock` CLI to acquire locks on the server side.
+    # Those use a "magic flag" to help indicate conflicts with semaphores.
+    # We want to compete with those locks though, so we use this flag too.
+    # Comes from https://github.com/hashicorp/consul/blob/master/api/lock.go#L35-L38
+    LOCK_MAGIC_FLAG = 0x2ddccbc058a50c18
```

Comment:
What sequence of event does this protect against?

Reply:
No sequence of events - it ensures we're playing the same game as the `consul lock` command.

Reply:
To quote the docs from the link there:

```go
	// LockFlagValue is a magic flag we set to indicate a key
	// is being used for a lock. It is used to detect a potential
	// conflict with a semaphore.
```

But we don't want it to think we're holding a semaphore.

Reply:
Yes I know, seems like hacks galore :P If we had budget we could do a lot better...

Reply:
Oh, **dang** I get it now. If not set [it will fail](https://github.com/hashicorp/consul/blob/master/api/lock.go#L198). Not sure how you figured that bit but well done :+1: